src/components/layout/TransactionWorkflow/TransactionWorkflow.tsx

```diff
@@ -80,30 +78,40 @@ function TransactionWorkflow({
         case 'next':
           switch (workflowStage) {
             case TRANSACTION_WORKFLOW_STATUS.PENDING: {
-              const newSteps = getWorkflowStepsForInteraction(interactionType);
-              newSteps['1'].status = 'success';
-              newSteps['2'].status = 'pending';
-              setSteps(newSteps);
+              steps['1'].status = 'success';
+              steps['2'].status = 'pending';
               dispatchTransactionState({
                 type: 'setWorkflowStage',
                 payload: TRANSACTION_WORKFLOW_STATUS.CONFIRMED,
               });
-              const originalTxId = await arweaveDataProvider.writeTransaction(
-                new ArweaveTransactionID(assetId),
-                {
-                  function: functionName,
-                  ...payload,
-                },
-              );
+              const originalTxId =
+                interactionType === INTERACTION_TYPES.CREATE &&
+                isObjectOfTransactionPayloadType<CreatePDNTPayload>(
+                  payload,
+                  TRANSACTION_DATA_KEYS[INTERACTION_TYPES.CREATE].keys,
+                )
+                  ? await arweaveDataProvider.deployContract({
+                      srcCodeTransactionId: new ArweaveTransactionID(
+                        payload.srcCodeTransactionId,
+                      ),
+                      initialState: payload.initialState,
+                      tags: payload?.tags,
+                    })
+                  : await arweaveDataProvider.writeTransaction(
+                      new ArweaveTransactionID(assetId),
+                      {
+                        function: functionName,
+                        ...payload,
+                      },
+                    );
               if (originalTxId) {
                 dispatchTransactionState({
                   type: 'setDeployedTransactionId',
-                  payload: originalTxId,
+                  payload: new ArweaveTransactionID(originalTxId.toString()),
                 });
               }
-              newSteps['2'].status = 'success';
-              newSteps['3'].status = 'success';
-              setSteps(newSteps);
+              steps['2'].status = 'success';
+              steps['3'].status = 'success';
```

Comment:
nit: put this in a function and call it from the switch statement. it's hard to understand what's going on in here with it nested

Reply:
added